cumulusci/tasks/bulkdata/load.py

```diff
@@ -188,8 +188,11 @@ def _stream_queried_data(self, mapping, local_ids, query):
             date_context = mapping.get_relative_date_context(
                 mapping.get_load_field_list(), self.org_config
             )
-
-        for row in query.yield_per(BULK_BATCH_SIZE):
+        # Clamping the yield from the query ensures we do not
+        # create more Bulk API batches than expected, regardless
+        # of batch size, while capping memory usage.
```

Comment:
I was going to ask if limiting the return from the sql database hurts performance, but then I remembered that's really not our bottleneck.

Reply:
For the record, I found this documentation in the git history. it's dreed's. I think I accidentally deleted it at some point.